bundles/binding/org.openhab.binding.zibase/src/main/java/org/openhab/binding/zibase/internal/ZibaseBinding.java

```diff
@@ -0,0 +1,247 @@
+/**
+ * Copyright (c) 2010-2014, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.zibase.internal;
+
+import java.util.Collection;
+import java.util.Dictionary;
+import org.openhab.binding.zibase.ZibaseBindingProvider;
+import org.openhab.binding.zibase.internal.ZibaseGenericBindingProvider;
+import org.openhab.binding.zibase.internal.ZibaseBindingConfig;
+import org.apache.commons.lang.StringUtils;
+import org.openhab.core.binding.AbstractActiveBinding;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import fr.zapi.Zibase;
+
+
+/**
+ * Implement this class if you are going create an actively polling service
+ * like querying a Website/Device.
+ * 
+ * @author Julien Tiphaine
+ * @since 1.6.0
+ */
+public class ZibaseBinding extends AbstractActiveBinding<ZibaseBindingProvider> implements ManagedService {
+
+	/**
+	 * generic logger
+	 */
+	private static final Logger logger = LoggerFactory.getLogger(ZibaseBinding.class);	
+	
+	/**
+	 * zibase to which we are connected
+	 */
+	private static Zibase zibase;
+	
+	/**
+	 * zibase binding provider
+	 */
+	private static ZibaseGenericBindingProvider bindingProvider = new ZibaseGenericBindingProvider();
+	
+	/**
+	 * Associated Zibase listener instance
+	 */
+	private static ZibaseListener zibaseListener = null; 
+	
+	/** 
+	 * the refresh interval which is used to poll values from the zibase
+	 * server (optional, defaults to 60000ms)
+	 */
+	private static long refreshInterval = 60000;
+	
+	/**
+	 * ip adresse of zibase to connect to
+	 */
+	private static String ip;
+	
+	/**
+	 * ip address sent to Zibase for registering
+	 */
+	private static String listenerHost = "127.0.0.1";
+	
+	/**
+	 * ip address sent to Zibase for registering
+	 */
+	private static int listenerPort = 9876;
+	
+	
+	/**
+	 * Constructor
+	 */
+	public ZibaseBinding() {
+
+	}
+	
+	/**
+	 * get the associated binding provider
+	 * @return
+	 */
+	public static ZibaseGenericBindingProvider getBindingProvider() {
+		return bindingProvider;
+	}
+		
+	/**
+	 * @{inheritDoc}
+	 */
+	public void activate() {
+		
+	}
+	
+	private void launch() {
+		// insure that lister is not yet running
+		if(zibaseListener!=null) {
+			zibaseListener.shutdown();
+		}
+		
+		// connect to zibase
+		try {
+			zibase = new Zibase(ip);
+			logger.info("connected to zibase for command sending");
+						
+		} catch(Throwable th)	{
```

Comment:
catching Throwable is never nice. Could you specify the catched Exception more specifically?